compiler/src/dotty/tools/dotc/util/ParsedComment.scala

```diff
@@ -0,0 +1,230 @@
+    "@since"   -> TagFormatter("Since", toMarkdownList),
+    "@version" -> TagFormatter("Version", toMarkdownList)
+  )
+
+  /**
+   * Formats a list of items into a list describing them.
+   *
+   * Each element is assumed to consist of a first word, which is the item being described. The rest
+   * is the description of the item.
+   *
+   * @param items The items to format into a list.
+   * @return A markdown list of descriptions.
+   */
+  private def toDescriptionList(ctx: Context, items: List[String]): String = {
+    val formattedItems = items.map { p =>
+      val name :: rest = p.split(" ", 2).toList
+      s"${bold(name)(ctx)} ${rest.mkString("").trim}"
+    }
+    toMarkdownList(ctx, formattedItems)
+  }
+
+  /**
+   * Formats a list of items into a markdown list.
+   *
+   * @param items The items to put in a list.
+   * @return The list of items, in markdown.
+   */
+  private def toMarkdownList(ctx: Context, items: List[String]): String = {
+    val formattedItems = items.map(_.lines.mkString(System.lineSeparator + "   "))
+    formattedItems.mkString(" - ", System.lineSeparator + " - ", "")
+  }
+
+  /**
+   * If the color is enabled, add syntax highlighting to each of `snippets`, otherwise wrap each
+   * of them in a markdown code fence.
+   * The results are put into a markdown list.
+   *
+   * @param language The language to use for the code fences
+   * @param snippets The list of snippets to format.
+   * @return A markdown list of code fences.
+   * @see toCodeFence
+   */
+  private def toCodeFences(language: String)(ctx: Context, snippets: List[String]): String =
+    toMarkdownList(ctx, snippets.map(toCodeFence(language)(ctx, _)))
+
+  /**
+   * Formats `snippet` for display. If the color is enabled, the syntax is highlighted,
+   * otherwise the snippet is wrapped in a markdown code fence.
+   *
+   * @param language The language to use.
+   * @param snippet  The code snippet
+   * @return `snippet`, wrapped in a code fence.
+   */
+  private def toCodeFence(language: String)(ctx: Context, snippet: String): String = {
+    if (colorEnabled(ctx)) {
+      SyntaxHighlighting.highlight(snippet)(ctx)
+    } else {
+      s"""```$language
+         |$snippet
+         |```""".stripMargin
+    }
+  }
+
+  /**
+   * Format the elements of documentation associated with a given tag using `fn`, and starts the
+   * section with `title`.
+   *
+   * @param title The title to give to the formatted items.
+   * @param fn    The formatting function to use.
+   */
+  private case class TagFormatter(title: String, fn: (Context, List[String]) => String) {
+
+    /**
+     * Format `item` using `fn` if `items` is not empty.
+     *
+     * @param items The items to format
+     * @return If items is not empty, the items formatted using `fn`.
+     */
+    def apply(items: List[String])(implicit ctx: Context): Option[String] = items match {
+      case Nil =>
+        None
+      case items =>
+        Some(s"""${heading(title)}:
```

Comment:
I would remove the ":" since it should be clear this is a title from the formatting.